### DISCUSSION TITLE
NioConnection Unable to connect to remote: is there a server running on port 8250

### DISCUSSION THREAD

**Discussion by logobackwoods** [apache/cloudstack#10092] [Q&A] [answered]
##### ISSUE TYPE

 * Bug Report
 * Other

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~
2024-12-11 09:44:23,125 INFO  [cloud.agent.Agent] (main:null) (logid:) Agent [id = 1 : type = LibvirtComputingResource : zone = 1 : pod = 1 : workers = 5 : host = 192.168.127.60 : port = 8250
2024-12-11 09:44:23,130 INFO  [utils.nio.NioClient] (main:null) (logid:) Connecting to 192.168.127.60:8250
2024-12-11 09:44:23,131 WARN  [utils.nio.NioConnection] (main:null) (logid:) Unable to connect to remote: is there a server running on port 8250
~~~

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by logobackwoods** [comment]
root@jachs:~# ss -tuln | grep 8250
root@jachs:~# 


root@jachs:~# systemctl status libvirtd
● libvirtd.service - libvirt legacy monolithic daemon
   Loaded: loaded (/usr/local/lib/systemd/system/libvirtd.service; enabled; vendor preset: enabled)
  Drop-In: /etc/systemd/system/libvirtd.service.d
           └─override.conf
   Active: active (running) since Wed 2024-12-11 09:49:08 UTC; 2s ago
     Docs: man:libvirtd(8)
           https://libvirt.org/
 Main PID: 5188 (libvirtd)
    Tasks: 20 (limit: 32768)
   CGroup: /system.slice/libvirtd.service
           └─5188 /usr/local/sbin/libvirtd --timeout 120

Dec 11 09:49:08 jachs systemd[1]: Starting libvirt legacy monolithic daemon...
Dec 11 09:49:08 jachs systemd[1]: Started libvirt legacy monolithic daemon.

**Comment by logobackwoods** [comment]
root@jachs:~# ping 192.168.127.60 
PING 192.168.127.60 (192.168.127.60) 56(84) bytes of data.
64 bytes from 192.168.127.60: icmp_seq=1 ttl=64 time=0.057 ms
64 bytes from 192.168.127.60: icmp_seq=2 ttl=64 time=0.039 ms
64 bytes from 192.168.127.60: icmp_seq=3 ttl=64 time=0.049 ms
64 bytes from 192.168.127.60: icmp_seq=4 ttl=64 time=0.045 ms
^C
--- 192.168.127.60 ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3053ms
rtt min/avg/max/mdev = 0.039/0.047/0.057/0.009 ms

root@jachs:~# telnet 192.168.127.60 8250
Trying 192.168.127.60...
telnet: Unable to connect to remote host: Connection refused
root@jachs:~# telnet 192.168.127.60 8080
Trying 192.168.127.60...
Connected to 192.168.127.60.
Escape character is '^]'.
^CConnection closed by foreign host.

root@jachs:~# ufw status
Status: inactive

**Comment by DaanHoogland** [comment]
@logobackwoods what kind of setup is this?
- a separate server and agent?
- KVM on Ubuntu?
- How did you do the setup?

thanks,

**Comment by logobackwoods** [comment]
@DaanHoogland I solved that problem. I have not received a service on port 8250. After the restart, the problem was resolved, but a new one appeared.

Dec 12 08:22:19 jachs java[29560]: ERROR [c.c.u.n.Link] (AgentManager-SSLHandshakeHandler-2:null) (logid:) SSL error caught during wrap data: Empty server certificate chain, for local address=<ip> , remote address=<ip>.
Dec 12 08:22:19 jachs java[34016]: ERROR [utils.nio.Link] (main:) (logid:) SSL error caught during unwrap data: Received fatal alert: bad_certificate, for local address=<ip>, remote address=<ip>. The client may have invalid ca-certificates.
Dec 12 08:22:19 jachs java[34016]: ERROR [utils.nio.NioClient] (main:) (logid:) SSL Handshake failed while connecting to host: <ip> port: 8250
Dec 12 08:22:19 jachs java[34016]: ERROR [utils.nio.NioConnection] (main:) (logid:) Unable to initialize the threads.

**Reply by verendrakalsi** [reply]
@logobackwoods  how did you resolved this issue ?
[utils.nio.NioConnection] (main:null) (logid:) Unable to connect to remote: is there a server running on port 8250

**Reply by rajujith** [reply, marked as answer]
@logobackwoods  looks like some error in the management server /agent installation. I think reinstalling could help.
